### PR TITLE
[Blog] Add the "Control deployment on Platform.sh" post

### PR DESCRIPTION
**Pull request by BR0kEN-**

### CODE REVIEW

File: docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+
+## What's the problem?
+
+The hosting gives you two steps for completing the deployment: [build](https://docs.platform.sh/configuration/app/build.html#build-hook) and [deploy](https://docs.platform.sh/configuration/app/build.html#deploy-hook). During the `build` you have a writable filesystem, but no services available (like MariaDB, Redis, RabbitMQ, etc.). At the `deploy` the FS is switched to read-only mode and all the services are ready for usage.
+
+The problem is completely tethered to the `deploy` operation and its technical description is the following: **unable to bubble up the exit code of deployment to handle it properly and mark process as failed.**
```
Comment (lokeoke) [inline docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md]: >problem

**SUGGESTION**
>issue

File: docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+
+The `build` hook executes outside of a container so any exit code can easily stop the process. That cannot be said about the `deploy` hook because its area of operation is within a container and there's no technical solution exists at desks of Platform.sh development team to pass it out to the level where it can be handled.
```
Comment (lokeoke) [inline docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md]: >The `build` hook executes

**SUGGESTION**
>The `build` hook is executed

File: docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+
+      $this->middleware->handle($this->request);
+    }
+
+  }
+  ```
+
+## Conclusion
+
+The Platform.sh is a good hosting with its container dedication and configuration flexibilities. However, as usual, there is a lot of work to do for improving the service.
```
Comment (sirko) [inline docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md]: >The Platform.sh is a good hosting with its container dedication and configuration flexibilities. 

**Suggestion**

> The Platform.sh provides a good service with unique container dedication and configuration flexibilities.

Comment (sirko) [inline docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md]: > However, as usual, there is a lot of work to do for improving the service.

**Suggestion**

>However their development team has enough space for improvements.

File: docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+
+I was also trying to use it as continuous integration tool but it really built in a way to stop you from doing this. For instance, you can't have Xdebug installed in some environments to generate PHPUnit coverage reports. If you are adding something, it'll be added everywhere. Everywhere means `master` environment too. The PHPUnit tests, PHPCS or HTMLCS checks etc. cannot be run during the `build` but can be at `deploy`. When the `build` has passed it means the application is ready for usage... I believe you've got the point - use dedicated CI instruments for achieving continuous delivery to Platform.sh and don't try to put everything to its shoulders. At the bottom line, it's just a hosting!
```
Comment (sirko) [inline docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md]: > ...but it really built in a way to stop you from doing this. 
**Suggestion**
>...but, unfortunately I found that it is not the best tool for this purpose.

Comment (sirko) [inline docs/_posts/2018-05-13-platform-sh-fail-on-deploy.md]: > At the bottom line, it's just a hosting!

It is not a hosting, but rather a service. They are describing themselves as " Platform-as-a-Service built especially for continuous deployment". 
But if you want to use good old **hosting** term, then do not forget that it is better to say **hosting provider** or **hosting company**

**Comment by BR0kEN-** [comment]
@sirko, @lokeoke, thanks for your reviews, chaps. Now it should look better.

By the way, to run the local copy of cikit.tools you can do the following:

```
git clone git@github.com:BR0kEN-/cikit.git --branch=blog/platform-sh-fail-on-deploy
cd cikit/docs
bundle install
bundle exec jekyll serve
```

Then the article will be available at http://localhost:4000/blog/platform-sh-fail-on-deploy/

**Pull request merged**